docs/tags_doc.py

```diff
@@ -0,0 +1,87 @@
+"""
+Parses required places in the cp-ansible project for all instances of tags in following format
+tags: <test-tag>
+tags:
+    - <test-tag>
+
+And writes the list of tags to a formatted markdown file.
+Adds new entries for new tags with blank description (to be filled by the user).
+For the existing entries with description, lets it stay as is.
+"""
+
+import os
+
+# Parse all tags in cp-ansible
+def get_all_tags(directory, search_text, filePattern):
+    
+    all_tags = []
+    for path, dirs, files in os.walk(os.path.abspath(directory)):
+        for filename in files:
+            filepath = os.path.join(path, filename)
+            if not filename.endswith(filePattern)  : 
+                continue
+            with open(filepath) as f:
+                file_content = f.read()
+            if search_text not in file_content:
+                continue
+            list_lines = file_content.split("\n")
+            i = 0
+            while (i < len(list_lines)):
+                line = list_lines[i]
+                i = i+1
+                if search_text in line:
+                    tag = line.replace(search_text,'').strip()
+                    if (tag == ''):
+                        while(i< len(list_lines) and "-" in list_lines[i]):
+                            new_tag = list_lines[i].replace("-", '').strip()
```

Comment:
Its not a scalable way to find the the tags. Tags are essentially list and list can be represented in more ways that what is handled here. i.e `tags: [tag1, tag2]`.
Please take a look at `list-tags` options in ansible. If that doesn't serve purpose, try to convert yaml to dict and let work on dict recursively rather than string manipulations.

Reply:
IMO the output of list-tags is more structured. On top of that you get the additional metadata of tags on pay vs tags on task. More importantly, you don't have to worry about the spaces, `tags:` in comments, multiline big array representation etc.

Reply:
I'm exploring the dict way. 
If that doesn't seem to work easily, we'll have to go back to file parsing and string manipulation

Reply:
That'd have been perfect. But it doesn't seem to work. 
For ex 
`ansible-playbook confluent.platform.all --list-tag` works but if I do
`cd playbooks; ansible-playbook all.yml --list-tag` or `ansible-playbook playbooks/all.yml --list-tag`, it fails. Reason is that it goes into recursive search but is not able to resolve other roles/tasks if we don't use the playbook in the collection form (i.e. as confluent.platform.all)
We can separately call all playbooks in playbooks directory. But we still will miss a few tags. 

I'm reading more to see if there's a way to run it on all yaml files without it doing any recursive check